### DISCUSSION TITLE
dependencies in plugin.xml

### DISCUSSION THREAD

**Discussion by moglideveloper** [JetBrains/sbt-idea-plugin#109] [Q&A]
Our <b>sample intellij plugin</b> is dependent on intellij scala plugin.   

So, as per our functional requirement,   
 <b>sample intellij plugin</b> supposed to work only on those versions of intellij and intellij products,   
where intellij scala plugin works.

---
---

# Below are 3 screenshots from page https://plugins.jetbrains.com/docs/intellij/plugin-compatibility.html#modules-specific-to-functionality :-

---
<img width="820" alt="image" src="https://user-images.githubusercontent.com/8999083/117903678-9dd66980-b2ed-11eb-8023-f4c9117f931e.png">

---
<img width="843" alt="image" src="https://user-images.githubusercontent.com/8999083/117903710-ae86df80-b2ed-11eb-9391-3357ffa64858.png">

---
<img width="863" alt="image" src="https://user-images.githubusercontent.com/8999083/117903751-c0688280-b2ed-11eb-959d-7e384e027c4b.png">

---
---

We are follwing recommendations as discussed here https://github.com/JetBrains/sbt-idea-plugin/discussions/105
and above, still we are not getting a clarity on dependent plugins.

We have only declared a dependency on **org.intellij.scala** and there is a warning in our plugin.xml :

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/8999083/117904403-efcbbf00-b2ee-11eb-9746-72e286cd5c35.png">

_**Kindly confirm, what are all the plugins that needs to be declared in plugin.xml,
if <b>sample intellij plugin</b> supposed to work only on those versions of intellij and intellij products,   
where intellij scala plugin works.**_

---

Please see :
```
    <depends>com.intellij.modules.platform</depends>
    <depends>com.intellij.modules.lang</depends>
```
https://github.com/AVSystem/intellij-hocon/blob/bd4314ff3d8dffc0a56bb314a978e2a81cb05d7c/resources/META-INF/plugin.xml#L9-L10

---

Please also see below comment for reference where **com.intellij.modules.platform**, **com.intellij.properties** are also declared in addition to **org.intellij.scala**:

and `plugin.xml`
```xml
<idea-plugin>
    <id>naumneko.test.vendor.testintellij2</id>
    <name>TestIntellij2</name>
    <version>0.0.1-SNAPSHOT</version>
    <vendor>naumneko.test.vendor</vendor>

    <description>A new plugin for IntelliJ platform written in Scala and built with SBT</description>

    <idea-version since-build="211.0"/>

    <depends>com.intellij.modules.platform</depends>
    <depends>com.intellij.properties</depends>
    <depends>org.intellij.scala</depends>
    
<!--    SOME ACTIONS, LISTENERS, etc... -->
</idea-plugin>

```

_Originally posted by @unkarjedy in https://github.com/JetBrains/sbt-idea-plugin/discussions/107#discussioncomment-722482_

---

Also see https://github.com/chathurabuddi/intellij-jasper-report-support/pull/5/files

**Comment by unkarjedy** [comment]
@mutcianm 
Can it be that DevKit plugin doesn't recognize `<!DOCTYPE ...>`?
I see it in Scala PLugin `plugin.xml` but not in other plugins.
BTW scala plugin is available in the completion list inside `<depends>` it's just resolve/goto that doesn't work on it's id, once it's completed

![image](https://user-images.githubusercontent.com/3989292/117938783-12ba9b00-b310-11eb-9634-87ea0ed47a46.png)

![image](https://user-images.githubusercontent.com/3989292/117938928-3978d180-b310-11eb-9090-60e07ab96856.png)

**Reply by niktrop** [reply]
@unkarjedy  I've removed doctype declaration from the scala plugin as well. It breaks parsing of plugin.xml in 2021.2.

**Reply by unkarjedy** [reply]
Maybe it would be safe to port into 2021.1, taking into account that other plugins in 211 don't have it?

**Reply by moglideveloper** [reply]
Upgraded to :
`intellijPlugins += "org.intellij.scala:2021.1.21:Eap".toPlugin`

Still, `<depends>org.intellij.scala</depends>` is in red.

**Kindly confirm, what are all the plugins that needs to be declared in plugin.xml,
if sample intellij plugin supposed to work only on those versions of intellij and intellij products,
where intellij scala plugin works.**

<img width="1111" alt="image" src="https://user-images.githubusercontent.com/8999083/119303250-0ecd3800-bc83-11eb-92a6-e92f0acac1ab.png">
